DependencyInjection/DoctrineExtension.php

```diff
@@ -143,29 +143,35 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
         $defitionClassname = $this->getDefinitionClassname();
 
         $configuration = $container->setDefinition(sprintf('doctrine.dbal.%s_connection.configuration', $name), new $defitionClassname('doctrine.dbal.connection.configuration'));
-        $logger = null;
+
+        $loggers = [];
         if ($connection['logging']) {
-            $logger = new Reference('doctrine.dbal.logger');
+            $loggers[] = new Reference('doctrine.dbal.logger');
         }
-        unset ($connection['logging']);
+        unset($connection['logging']);
+
         if ($connection['profiling']) {
             $profilingLoggerId = 'doctrine.dbal.logger.profiling.'.$name;
             $container->setDefinition($profilingLoggerId, new $defitionClassname('doctrine.dbal.logger.profiling'));
+
             $profilingLogger = new Reference($profilingLoggerId);
             $container->getDefinition('data_collector.doctrine')->addMethodCall('addLogger', array($name, $profilingLogger));
 
-            if (null !== $logger) {
-                $chainLogger = new $defitionClassname('doctrine.dbal.logger.chain');
-                $chainLogger->addMethodCall('addLogger', array($profilingLogger));
+            $loggers[] = new Reference($profilingLoggerId);
+        }
+        unset($connection['profiling']);
 
-                $loggerId = 'doctrine.dbal.logger.chain.'.$name;
-                $container->setDefinition($loggerId, $chainLogger);
-                $logger = new Reference($loggerId);
-            } else {
-                $logger = $profilingLogger;
+        if (!empty($loggers)) {
+            $chainLogger = new $defitionClassname('doctrine.dbal.logger.chain');
+            foreach ($loggers as $logger) {
```

Comment:
if we have only 1 logger, we should not wrap it in a chain. No need to add overhead for nothing.

Reply:
Wrapping was done to make it easier to add more loggers, overhead should be non-existent, but I will revert this so that I don't change `DoctrineExtension` at all

Reply:
Overhead is going through the chain logger layer for each logging call, which may be a lot if you have lots of DB queries (I agree that the cost of the DB query outweights the logger cost anyway though).
Making the compiler pass deal with an existing logger set in the configuration already would be better